app/src/main/java/org/hwyl/sexytopo/control/util/Space2DUtils.java

```diff
@@ -94,4 +96,57 @@ public static double adjustAngle(double angle, double delta) {
         Coord2D end = line.getEnd();
         return new Line<>(start.plus(point), end.plus(point));
     }
+
+    private static List<Coord2D> DouglasPeukerIteration(List<Coord2D> path, double epsilon) {
+
+        // Find the point with the maximum distance
+        int pathSize = path.size();
+        int indexMax = 0;
+        double distMax = 0;
+
+        for (int i = 1; i != pathSize; ++i) {
+            double dist = getDistanceFromLine(path.get(i), path.get(0), path.get(pathSize - 1));
+            if (dist > distMax) {
+                distMax = dist;
+                indexMax = i;
+            }
+        }
+
+        List<Coord2D> simplifiedPath;
+
+        // If max distance is greater than epsilon, recursively simplify
+        if (distMax > epsilon) {
+            List<Coord2D> results1 = Simplify(path.subList(0, indexMax + 1), epsilon);
+            List<Coord2D> results2 = Simplify(path.subList(indexMax, pathSize), epsilon);
+
+            simplifiedPath = new ArrayList<>(results1);
+            simplifiedPath.addAll(results2.subList(1, results2.size()));
+        } else {
+            simplifiedPath = new ArrayList<>();
+            simplifiedPath.add(path.get(0));
+            simplifiedPath.add(path.get(pathSize - 1));
+        }
+
+        return simplifiedPath;
+    }
+
+    public static double SketchEpsilon(double width, double height) {
+
+        // TODO: this is a pretty crude simplification factor but we can revisit after more testing
+        double maxDim = Math.max(width, height);
+        return Math.max(maxDim / 500, 0.001);
+    }
+
+    public static double SketchEpsilon(PathDetail pathDetail) {
+        return SketchEpsilon(pathDetail.getWidth(), pathDetail.getHeight());
+    }
+
+    public static List<Coord2D> Simplify(List<Coord2D> path, double epsilon) {
+
+        if (path.isEmpty() || epsilon <= 0)
+            return path;
+
+        return DouglasPeukerIteration(path, epsilon);
+    }
+
```

Comment:
These methods may need moving to more appropriate homes but I'll leave that decision to you :)

Reply:
Fair enough - not a terrible place for them to live for now. Although they should have lowercase first letters because the initial caps looks a bit weird in Java.